go-controller/pkg/node/gateway_iptables.go

```diff
@@ -467,6 +467,37 @@ func getLocalGatewayNATRules(ifname string, cidr *net.IPNet) []nodeipt.Rule {
 			Protocol: protocol,
 		},
 	}
+	if util.IsNetworkSegmentationSupportEnabled() {
+		// NOTE: Ordering is important here, the RETURN must come before
+		// the MASQUERADE rule. Please don't change the ordering.
+		srcUDNMasqueradePrefix := config.Gateway.V4MasqueradeSubnet
+		defaultNetworkReservedMasqueradePrefix := config.Gateway.MasqueradeIPs.V4HostMasqueradeIP.String() + "/29"
+		if protocol == iptables.ProtocolIPv6 {
+			srcUDNMasqueradePrefix = config.Gateway.V6MasqueradeSubnet
+			defaultNetworkReservedMasqueradePrefix = config.Gateway.MasqueradeIPs.V6HostMasqueradeIP.String() + "/125"
+		}
+		rules = append(rules,
```

Comment:
have you thought about jumping to a new chain and adding these rules? It would allow us to handle the stale rules scenario.
You can use iptables managers "OwnChain" then to ensure no stales.
Not a hard ask - just an idea.

Reply:
Sure @tssurya